<launch>

    <arg name="scan_topic" default="scan" />                                                    <!-- Name of lidar scan topic -->
    <arg name="pattern_topic" default="pattern" />                                              <!-- Name of lidar pattern topic -->
    <arg name="clustered_points_topic" default="clustered_points" />                            <!-- Name of lidar clustered point topic -->
    <arg name="pattern_filepath" default="$(find pattern_matcher)/pcd/pattern.pcd" />           <!-- Filepath of pcd file -->

    <node name="pattern_matcher" type="pattern_matcher" pkg="pattern_matcher" output="screen">
        <param name="scan_topic" value="$(arg scan_topic)" />
        <param name="pattern_topic" value="$(arg pattern_topic)" />
        <param name="clustered_points_topic" value="$(arg clustered_points_topic)" />
        <param name="pattern_filepath" value="$(arg pattern_filepath)" />
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find pattern_matcher)/config/pattern_matcher.rviz"/>

</launch>